task_processing/registry.py

```diff
@@ -0,0 +1,93 @@
+import importlib
+
+import pyrsistent
+
+from task_processing.interfaces import TaskExecutor
+
+
+class Registry(object):
+    """Registry for Task Processing plugins
+
+    This is the entrypoint to registering plugins (which know how to
+    actually run tasks) and creating instances of TaskExecutors. Note that
+    since most customers of this library will be importing this class into
+    their code, let's try to make it Python2/3 compatible
+
+    Typical use would be:
+
+        reg = Registry()
+        # registers 'myplugin'
+        reg.load_plugin('my.plugin')
+        # ... within the my.plugin.register_plugin method:
+        registry.add_task_executor('foobar', FooBarExecutor)
+
+        # register 'another'
+        reg.load_plugin('another.plugin')
+
+        # Later...
+        executor = reg.executor_from_config(
+            provider='foobar',
```

Comment:
I do want plugins to be able to supply multiple executors, but prefixing them with the plugin name (and making that the package name) would have the nice side effect of making sure you can't have two of the same name ... that's actually pretty nice.

I will change this tomorrow to prefix all registered executors with the package name (which is the plugin name)?